src/main/java/org/squiddev/plethora/integration/hatchery/IntegrationHatchery.java

```diff
@@ -0,0 +1,287 @@
+			//Check for a captured entity
+			NBTTagCompound nbt = context.getTarget().getTagCompound();
+			if (nbt == null || !nbt.hasKey("storedEntity")) return Collections.emptyMap();
+
+			//Check for an entity ID
+			NBTTagCompound entityData = nbt.getCompoundTag("storedEntity");
+			if (!entityData.hasKey("id", Constants.NBT.TAG_STRING)) return Collections.emptyMap();
+
+			//Check if we have a location
+			IWorldLocation location = context.getContext(ContextKeys.ORIGIN, IWorldLocation.class);
+			if (location == null) return getBasicDetails(entityData);
+
+			//Due to Hatchery's `ItemStackEntityNBTHelper` having a `saveAll` param, we may need to
+			// handle a generic egg.  Unfortunately, they don't provide a convenient analogue to TE's `ItemMorb.GENERIC`,
+			// so this will have to suffice...
+			Entity entity = EntityList.createEntityFromNBT(entityData, location.getWorld());
+			if (entity == null)
+				entity = EntityList.createEntityByIDFromName(new ResourceLocation(entityData.getString("id")), location.getWorld());
+			if (entity == null) return getBasicDetails(entityData);
+
+			Vec3d loc = location.getLoc();
+			entity.setPositionAndRotation(loc.x, loc.y, loc.z, 0, 0);
+			return Collections.singletonMap("spawnEntity", context.makePartialChild(entity).getMeta());
+		}
+
+		//Copied wholesale from IntegrationThermalExpansion.META_MORB
+		private Map<Object, Object> getBasicDetails(NBTTagCompound entityData) {
+			String translationKey = EntityList.getTranslationName(new ResourceLocation(entityData.getString("id")));
+			if (translationKey == null) return Collections.emptyMap();
+
+			String translated = Helpers.translateToLocal("entity." + translationKey + ".name");
+
+			Map<Object, Object> details = new HashMap<>(2);
+			details.put("name", translated);
+			details.put("displayName",
+				entityData.hasKey("CustomName", Constants.NBT.TAG_STRING)
+					? entityData.getString("CustomName")
+					: translated
+			);
+			return Collections.singletonMap("spawnEntity", details);
+		}
+
+		@Nullable
+		@Override
+		public ItemStack getExample() {
+			//REFINE What are the trade-offs between this technique and `new ResourceLocation("minecraft:chicken")`?
+			ResourceLocation chickenLocation = EntityList.getKey(EntityChicken.class);
+			if (chickenLocation == null) return null;
+
+			//REFINE Similarly, how does this compare with `new EntityChicken(WorldDummy.INSTANCE)`
+			Entity chicken = EntityList.createEntityByIDFromName(chickenLocation, WorldDummy.INSTANCE);
+			if (!(chicken instanceof EntityAgeable)) return null;
+
+			return HatcheryEgg.createEggFromEntity(WorldDummy.INSTANCE, (EntityAgeable) chicken);
+		}
+	};
+
+	public static final IMetaProvider<EntityRooster> META_ROOSTER = new BaseMetaProvider<EntityRooster>() {
+
+		@Nonnull
+		@Override
+		public Map<Object, Object> getMeta(@Nonnull IPartialContext<EntityRooster> context) {
+			Map<Object, Object> out = new HashMap<>();
```

Comment:
If you're only operating on the target. maybe use a `BasicMetaProvider` instead. Might also be worth using `Collections.singletonMap` as well, as there's only one item.

Reply:
Done, pending commit